src/store/broadcast.ts

```diff
@@ -1,22 +1,23 @@
 import type { Store } from 'redux'
-import type { Middleware, PreloadedState } from '@reduxjs/toolkit'
+import type { Middleware } from '@reduxjs/toolkit'
 import type { RootState } from '@/store'
 
-type PreloadedRootState = PreloadedState<RootState>
-
 const BC_NAME = 'SAFE__store-updates'
 const tabId = Math.random().toString(32).slice(2)
 let broadcast: BroadcastChannel | undefined
 
-export const broadcastState = <K extends keyof PreloadedRootState>(sliceNames: K[]): Middleware<{}, RootState> => {
-  return (_) => (next) => (action) => {
+export const broadcastState = <K extends keyof RootState>(sliceNames: K[]): Middleware<{}, RootState> => {
+  return (_) => (next) => (action: unknown) => {
     const result = next(action)
 
     // Broadcast actions that aren't being already broadcasted
-    if (!action._isBroadcasted) {
-      const sliceType = action.type.split('/')[0]
-      if (sliceNames.includes(sliceType)) {
-        broadcast?.postMessage({ action, tabId })
+    if (typeof action === 'object' && action !== null) {
+      const actionObj = action as { _isBroadcasted?: boolean; type?: string }
+      if (!actionObj._isBroadcasted && actionObj.type) {
+        const sliceType = actionObj.type.split('/')[0]
+        if (sliceNames.includes(sliceType as K)) {
```

Comment:
Was it doing more broadcasts than needed before?

Reply:
Those changes are more due to the new ts types. I had to silence ts. I don't think that there was an error in the previous code.